### ISSUE TITLE
Ticket file digests are different between two time builds

### ISSUE THREAD

**Issue by MaggieYingYi** [SNSystems/llvm-prepo#30]
Compile the following code:
```
template <typename _CharT>
class basic_ostream;

typedef basic_ostream<char> ostream;
template <typename _CharT>
basic_ostream<_CharT> &__ostream_insert(basic_ostream<_CharT> &__out,
                                        _CharT *__s);
inline basic_ostream<char> &operator<<(basic_ostream<char> &__out, char *__s) {
  __ostream_insert(__out, __s);
}

ostream &operator<<(ostream &stream, int version) {	
  char *str = "";
  switch (version) {
  case 1:
    str = "time_based";
    break;
  case 2:
    str;
  case 3:
    str = "name_based_md5";
  }
  return stream << str;
}

```
using
```
$ rm clang.db
$ clang++ -target x86_64-pc-linux-gnu-repo -O2 -std=gnu++11 -c test.cpp
$ clang++ -target x86_64-pc-linux-gnu-repo -O2 -std=gnu++11 -c test.cpp

```
The list of generations in the repository is:

```
- file :
      path : clang.db
      size : 4194304
  log  :
      - { number: 2, size: 208, time: 2018-03-01T09:36:37Z }
      - { number: 1, size: 1176, time: 2018-03-01T09:36:36Z }
      - { number: 0, size: 0, time: 2018-03-01T09:36:36Z }
...
```
During the second time build, there is no database transaction since the source file keeps the same. The latest revision number should be 1 instead of 2.

**Comment by MaggieYingYi** [comment]
The TicketNode metadata, which is created by LLVM backend, contributes to the ticket file digest calculation during the first time build. When compiling the same file again, there is no TicketNode metadata which is generated by the LLVM backend. Therefore, the ticket file digest value is different from the first time value.

To fix the issue, the TicketNode metadata which is created by the LLVM backed doesn’t contribute to the ticket file digest calculation.

**Comment by MaggieYingYi** [comment]
Fixed in git commit: 79b5d5fe2b714
